lib/request.js

```diff
@@ -624,7 +647,7 @@ internals.Info = class {
         this._request = request;
 
         const req = request.raw.req;
-        const host = req.headers.host ? req.headers.host.trim() : '';
+        const host = (req.headers.host || req.headers[':authority'] || '').trim();
```

Comment:
Perhaps we could test if the req is an IPv6 request somehow, and choose `Host` or `:authority` accordingly. WDYT?